requirements/Makefile

```diff
@@ -15,3 +17,5 @@ upgrade-requirements:
 	pip-compile --upgrade -o test-requirements.txt test-requirements.in --allow-unsafe
 	pip-compile --upgrade -o dev-requirements.txt dev-requirements.in --allow-unsafe
 	pip-compile --upgrade -o docs-requirements.txt docs-requirements.in --allow-unsafe
+	awk 'BEGIN { del=0 } /appnope/ { del=1 } del<=0 { print } /via ipython/ { del -= 1 }' prod-requirements.txt > temp-prod-requirements.txt && mv temp-prod-requirements.txt prod-requirements.txt
+	awk 'BEGIN { del=0 } /appnope/ { del=1 } del<=0 { print } /via ipython/ { del -= 1 }' dev-requirements.txt > temp-dev-requirements.txt && mv temp-dev-requirements.txt dev-requirements.txt
```

Comment:
Would be nice to eliminate this duplication. A separate Makefile target should work fine here (I think). For example:

```Makefile
.PHONY: requirements upgrade-requirements purge-appnope

purge-appnope: *-requirements.txt
	@echo "check/purge appnope:"
	@for reqtxt in $^; do \
		echo "- $$reqtxt"; \
		sed -E -i 'N;/^appnope=.+\n    # via ipython$$/d' "$${reqtxt}"; \
	done

# snip ...

	pip-compile -o docs-requirements.txt docs-requirements.in --allow-unsafe
	$(MAKE) purge-appnope

# snip ...

	pip-compile --upgrade -o docs-requirements.txt docs-requirements.in --allow-unsafe
	$(MAKE) purge-appnope



```

Reply:
**Update**: Just realized the above glob pattern didn't match the `requirements.txt` (no suffix) file.  Fixed in above code block. Change (`./` added for clarity):
```diff
-purge-appnope: *-requirements.txt
+purge-appnope: ./*requirements.txt
```